sdk/python/feast/infra/online_stores/contrib/mysql.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import absolute_import
+
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple
+
+import pymysql
+import pytz
+from pydantic import StrictStr
+from pymysql.connections import Connection
+
+from feast import Entity, FeatureView, RepoConfig
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.repo_config import FeastConfigBaseModel
+
+
+class MySQLOnlineStoreConfig(FeastConfigBaseModel):
+    """
+    Configuration for the MySQL online store.
+    NOTE: The class *must* end with the `OnlineStoreConfig` suffix.
+    """
+
+    type = "mysql"
+
+    host: Optional[StrictStr] = None
+    user: Optional[StrictStr] = None
+    password: Optional[StrictStr] = None
+    database: Optional[StrictStr] = None
+    port: Optional[int] = None
+
+
+class MySQLOnlineStore(OnlineStore):
+    """
+    An online store implementation that uses MySQL.
+    NOTE: The class *must* end with the `OnlineStore` suffix.
+    """
+
+    _conn: Optional[Connection] = None
+
+    def _get_conn(self, config: RepoConfig) -> Connection:
+
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, MySQLOnlineStoreConfig)
+
+        if not self._conn:
+            self._conn = pymysql.connect(
+                host=online_store_config.host or "127.0.0.1",
+                user=online_store_config.user or "test",
+                password=online_store_config.password or "test",
+                database=online_store_config.database or "feast",
+                port=online_store_config.port or 3306,
+                autocommit=True,
+            )
+        return self._conn
+
+    def online_write_batch(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        data: List[Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+
+        conn = self._get_conn(config)
+        cur = conn.cursor()
+
+        project = config.project
+
+        for entity_key, values, timestamp, created_ts in data:
+            entity_key_bin = serialize_entity_key(entity_key).hex()
```

Comment:
this should include `entity_key_serialization_version=config.entity_key_serialization_version`, see [here](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/online_stores/sqlite.py#L97) for an example

same for usage of `serialize_entity_key` below in `online_read`